data/scenarios/Challenges/Ranching/gated-paddock.yaml

```diff
@@ -0,0 +1,165 @@
+version: 1
+name: Pastoral Island
+description: |
+  Start a wool industry from the local fauna.
+creative: false
+objectives:
+  - goal:
+      - |
+        You've homesteaded on a small Pacific island.
+        It's time to gather resources to trade.
+      - |
+        You encounter some feral sheep. They look like a promising
+        source of wool.
+      - |
+        First, paddock the sheep so they don't drown.
+        Construct a fence using the nearby gate as a starting point.
+        Make sure there are no gaps!
+      - |
+        Note that you can use the "drill" command (by way of the "post puller" tool)
+        to demolish a fence that has been "placed".
+    condition: |
+      r <- robotnamed "sheep";
+      as r {run "scenarios/Challenges/Ranching/gated-paddock/enclosure-checking.sw"};
+robots:
+  - name: base
+    dir: [0, 1]
+    devices:
+      - treads
+      - scanner
+      - dictionary
+      - branch predictor
+      - strange loop
+      - clock
+      - ADT calculator
+      - comparator
+      - workbench
+      - lodestone
+      - grabber
```

Comment:
I suppose it's up to the player whether they want to construct a harvester.  They can "choose their own adventure"---weighing the effort of constructing a harvester vs. short term progress.

Reply:
I definitely made a harvester since it was not at all clear to me whether I would have enough trees otherwise, especially since I did not know the recipe for a fence.  I can see now that you included plenty of trees for making the fence, but even if I had known that, I was not sure what else I might need the trees for in the future.  Actually, I think having to make a little tree farm seems very much in keeping with the "homesteading" idea.  I wonder if you might consider starting with just a small stand of trees (like 6 or 7) and *requiring* the player to use a `harvester`.

Reply:
IMO this would not be possible in time to save all sheep.

What do you think, @kostmo?